components/FileUploadInput.tsx

```diff
@@ -0,0 +1,52 @@
+// TODO form action and submit as no-js fallback
+import React, { useRef } from "react";
+
+type Props = {
+  onFileSelected: (fileObj: File) => void
+}
+
+export default function FileUploadInput({ onFileSelected }: Props) {
+  const inputFile = useRef<HTMLInputElement>(null);
+
+  const onOpenFileInput = (e: React.MouseEvent) => {
+    inputFile.current?.click();
+    e.preventDefault();
+  }
+
+  const onHandleKeyPress = (e: React.KeyboardEvent) => {
+    if(e.key === 'Enter'){
+      e.preventDefault();
+      inputFile.current?.click();
+    }
+  }
+
+  const onFileChange = (e: React.FormEvent<HTMLInputElement>) => {
+    const target = (e.target as HTMLFormElement)
+    const fileObj = target.files[0];
+    onFileSelected(fileObj);
+
+    e.preventDefault();
```

Comment:
Nitpicky but I'd just put all of your `e.preventDefault` calls first

Reply:
Out of curiosity, is there a particular benefit to doing that?

Reply:
Code-wise, no - it's 100% for readability.

<details>
<summary>Overly-long explanation</summary>
For instance if you take these two methods

```
  const onOpenFileInput = (e: React.MouseEvent) => {
    inputFile.current?.click();
    e.preventDefault();
  }

  const onHandleKeyPress = (e: React.KeyboardEvent) => {
    if(e.key === 'Enter'){
      e.preventDefault();
      inputFile.current?.click();
    }
  }
```

You could normalize them to

```
  const onOpenFileInput = (e: React.MouseEvent) => {
    e.preventDefault();
    inputFile.current?.click();
  }

  const onHandleKeyPress = (e: React.KeyboardEvent) => {
    if(e.key !== 'Enter') return;
    e.preventDefault();
    inputFile.current?.click();
  }
```

which might lead you to refactoring it to

```
  const clickInput = (e) => {
    e.preventDefault();
    inputFile.current?.click();
  }

  const onOpenFileInput = clickInput;

  const onHandleKeyPress = (e: React.KeyboardEvent) => {
    if(e.key !== 'Enter') return;
    clickInput(e);
  }
```

It's a little contrived since it's only a handful of lines of code, but this sorta example is why I really push to keep code uniform. And in my not-so-humble opinion, uniform code implicitly says "I had a plan when writing this" vs "oh I forgot to add `preventDefault` the first time I tested this function"

I think it totally depends on the interviewer whether they'd care about something like that, but I figured it was worth mentioning anyway ✌️ 
</details>

Reply:
Cool cool, thanks for the answer!